test/src/helpers/pointsTest.cpp

```diff
@@ -0,0 +1,166 @@
+#include <points.hpp>
+#include <constans.hpp>
+#include <transformations.hpp>
+
+class PointCloudTest : public ::testing::TestWithParam<int>
+{
+protected:
+	std::vector<rgl_field_t> allNotDummyFields = {XYZ_F32,     IS_HIT_I32,  RAY_IDX_U32,  ENTITY_ID_I32,  INTENSITY_F32,
+	                                              RING_ID_U16, AZIMUTH_F32, DISTANCE_F32, RETURN_TYPE_U8, TIME_STAMP_F64};
+
+	std::vector<rgl_field_t> fieldsWithPaddings = {PADDING_32, XYZ_F32,       PADDING_8, IS_HIT_I32,
+	                                               PADDING_16, INTENSITY_F32, PADDING_32};
+
+	std::vector<rgl_field_t> fields = {XYZ_F32, IS_HIT_I32, INTENSITY_F32};
+
+	struct TestPointStruct
+	{
+		Field<XYZ_F32>::type xyz;
+		Field<IS_HIT_I32>::type isHit;
+		Field<INTENSITY_F32>::type intensity;
+	};
+
+	std::vector<TestPointStruct> generateTestPointsArray(int count)
+	{
+		std::vector<TestPointStruct> points;
+		points.reserve(count);
+		for (int i = 0; i < count; ++i) {
+			auto currentPoint = TestPointStruct{
+			    .xyz = {i, i + 1, i + 2},
+                  .isHit = i % 2, .intensity = 100
+            };
+			points.push_back(currentPoint);
```

Comment:
You can merge those lines, cuz emplace_back can take constructor arguments. 
Like previous:
U can construct on the flight, instead of constructing in place and moving.